src/Http/Http.Abstractions/src/ProblemDetails/ProblemDetailsContext.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Mvc;
+
+namespace Microsoft.AspNetCore.Http;
+
+/// <summary>
+/// Represent the current problem detatils context for the request.
+/// </summary>
+public sealed class ProblemDetailsContext
+{
+    /// <summary>
+    /// Creates a new instance of the <see cref="ProblemDetailsContext"/>.
+    /// </summary>
+    /// <param name="httpContext">The <see cref="HttpContext"/> associated with the current request being processed by the filter.</param>
+    public ProblemDetailsContext(HttpContext httpContext)
+    {
+        ArgumentNullException.ThrowIfNull(httpContext);
+
+        HttpContext = httpContext;
+    }
+
+    /// <summary>
+    /// The <see cref="HttpContext"/> associated with the current request being processed by the filter.
+    /// </summary>
+    public HttpContext HttpContext { get; }
+
+    /// <summary>
+    /// A collection of additional arbitrary metadata associated with the current request endpoint.
+    /// </summary>
+    public EndpointMetadataCollection? AdditionalMetadata { get; init; }
+
+    /// <summary>
+    /// A instance of <see cref="ProblemDetails"/> that will be
+    /// used during the response payload generation.
+    /// </summary>
+    public ProblemDetails ProblemDetails { get; init; } = new ProblemDetails();
```

Comment:
Does this instance (`new ProblemDetails()`) ever get used since the property is `init`?
```suggestion
    public ProblemDetails ProblemDetails { get; init; }
```

Reply:
I think it does make sense to be there because you can use it like this:

``` c#
var context = new ProblemDetailsContext(context);
context.ProblemDetails.Detail = "Custom Detail";
```

Removing it, means we will need to change it to `ProblemDetails?`. I will talk about that during the API Review. Thanks

Reply:
How about making it lazy?
```suggestion
    private ProblemDetails _problemDetails;
    public ProblemDetails ProblemDetails 
    {
        get
        {
            return (_problemDetails ??= new ProblemDetails())
        }
        init
        {
            _problemDetails = value;
        }
```

 This way the API does not need to change to `ProblemDetails?` and there won't be an extra instance of it here.
https://github.com/dotnet/aspnetcore/blob/72b97a1e0bb63d83786e43831a9b4feed163df4e/src/Mvc/Mvc.Core/src/Infrastructure/ObjectResultExecutor.cs#L124-L128